backend/src/center/guideViewProvider.ts

```diff
@@ -35,12 +36,18 @@ export default class GuideViewProvider implements vscode.WebviewViewProvider {
 						vscode.commands.executeCommand('loadGuidedDevelopment', data.value);
 						break;
 					}
+				case 'onInitialized':
+					{
+						this.refreshData(data.value.guides);
+						break;
+					}
 			}
-		});
+		}, this);
 	}
 
-	public refreshData() {
+	public refreshData(currentGuides?: any[]) {
 		let collections = Contributors.getInstance().getCollectionsInfo();
+		if (currentGuides && _.isEqual(currentGuides, collections)) return;
```

Comment:
According to the signature, the `collections` here might not be falsy values so that the truthy checking of `currentGuides` in first part could be optional?

Reply:
currentGuides is an optional parameter of this function, so we need to check it.

Reply:
> currentGuides is an optional parameter of this function, so we need to check it.

Yes, noticed that. I was referring the the equality between it and `currentGuides` since the later one is always an array (so undefined would not be equal to an array).

Reply:
Yes, ideally collections should be an array.
However, considering the completeness of the current function and performance(when currentGuides is undefined), it's better to add the check.